Reporting/ad-extensions-reporter.js

```diff
@@ -0,0 +1,395 @@
+            spreadsheet.deleteSheet(sheets[i]);
+        }
+    }
+
+    // Get report
+    Logger.log("Getting report");
+    if (typeof includeExtensionsWithZeroImpressions === 'undefined') {
+        Logger.log("Warning: options variable includeExtensionsWithZeroImpressions not found. " +
+            "Continuing, and by default including extensions that have had no impressions today."
+        );
+        impressionsCondition = '';
+    } else {
+        if (includeExtensionsWithZeroImpressions === true) {
+            var impressionsCondition = '';
+        } else if (includeExtensionsWithZeroImpressions === false) {
+            var impressionsCondition = 'WHERE Impressions > 0 ';
+        } else {
+            throw ("Value of options variable includeExtensionsWithZeroImpressions must be a boolean, either true or false. " +
+                "Currently, it is: " + includeExtensionsWithZeroImpressions)
+        }
+    }
+
+    var placeholderFeedItemReport = AdsApp.report(
+        'SELECT PlaceholderType, AttributeValues, DisapprovalShortNames ' +
+        'FROM PLACEHOLDER_FEED_ITEM_REPORT ' +
+        impressionsCondition +
+        'DURING TODAY'
+    );
+    var reportRows = placeholderFeedItemReport.rows();
+
+    // Process report data, to prepare for outputting to the spreadsheet
+    Logger.log("Processing report data");
+    var extensionPlaceholderTypeIds = Object.keys(extensionPlaceholderTypeIdToFieldIdsObjectMapping);
+
+    var extensionTypeFeedItemCounts = {};
+    for (var i = 0, n = extensionPlaceholderTypeIds.length; i < n; i++) {
+        extensionTypeFeedItemCounts[extensionPlaceholderTypeIds[i]] = 0;
+    }
+    var jsonAttributeValuesByExtensionType = {};
+    var disapprovalShortNamesByExtensionType = {};
+    var duplicateAttributeValuesCounts = {};
+
+    while (reportRows.hasNext()) {
+        var row = reportRows.next();
+
+        var placeholderTypeId = row['PlaceholderType']
+        if (extensionPlaceholderTypeIds.indexOf(placeholderTypeId) === -1) {
+            continue;
+        }
+        if (Object.keys(jsonAttributeValuesByExtensionType).indexOf(placeholderTypeId) === -1) {
+            jsonAttributeValuesByExtensionType[placeholderTypeId] = [];
+            disapprovalShortNamesByExtensionType[placeholderTypeId] = [];
+        }
+
+        var jsonAttributeValues = row['AttributeValues'];
+        var jsonDisapprovalShortNames = row['DisapprovalShortNames'];
+        if ('' === jsonDisapprovalShortNames.toString()) {
+            var disapprovalShortNames = [];
+        } else {
+            var disapprovalShortNames = JSON.parse(jsonDisapprovalShortNames);
+        };
+        if (jsonAttributeValuesByExtensionType[placeholderTypeId].indexOf(jsonAttributeValues) === -1) {
+            jsonAttributeValuesByExtensionType[placeholderTypeId].push(jsonAttributeValues);
+            disapprovalShortNamesByExtensionType[placeholderTypeId].push(disapprovalShortNames);
```

Comment:
Here you could add `duplicateAttributeValuesCounts[jsonAttributeValues] = 1`, then you wouldn't need the 2 check on like 282, or the 1 check on line 322

Reply:
Yep, nice!